test_vm/tests/publish_deals_test.rs

```diff
@@ -480,6 +489,87 @@ fn psd_all_deals_are_bad() {
     v.assert_state_invariants();
 }
 
+#[test]
+fn psd_bad_sig() {
+    let store = MemoryBlockstore::new();
+    let (v, a, deal_start) = setup(&store);
+    let (storage_price_per_epoch, provider_collateral, client_collateral) = token_defaults();
+
+    let deal_label = "deal0".to_string();
+    let proposal = DealProposal {
+        piece_cid: make_piece_cid(deal_label.as_bytes()),
+        piece_size: PaddedPieceSize(1 << 30),
+        verified_deal: false,
+        client: a.client1,
+        provider: a.maddr,
+        label: Label::String(deal_label),
+        start_epoch: deal_start,
+        end_epoch: deal_start + DEAL_LIFETIME,
+        storage_price_per_epoch,
+        provider_collateral,
+        client_collateral,
+    };
+
+    let invalid_sig_bytes = "very_invalid_sig".as_bytes().to_vec();
+    let publish_params = PublishStorageDealsParams {
+        deals: vec![ClientDealProposal {
+            proposal: proposal.clone(),
+            client_signature: Signature {
+                sig_type: SignatureType::BLS,
+                bytes: invalid_sig_bytes.clone(),
+            },
+        }],
+    };
+
+    let ret = v
+        .apply_message(
+            a.worker,
+            *STORAGE_MARKET_ACTOR_ADDR,
+            TokenAmount::zero(),
+            MarketMethod::PublishStorageDeals as u64,
+            publish_params,
+        )
+        .unwrap();
+    assert_eq!(ExitCode::USR_ILLEGAL_ARGUMENT, ret.code);
+    let auth_params = AuthenticateMessageParams {
+        signature: invalid_sig_bytes,
+        message: serialize(&proposal, "deal proposal").unwrap().to_vec(),
+    };
+    let auth_params_ser = serialize(&auth_params, "auth params").unwrap();
+
+    ExpectInvocation {
+        to: *STORAGE_MARKET_ACTOR_ADDR,
+        method: MarketMethod::PublishStorageDeals as u64,
+        subinvocs: Some(vec![
+            ExpectInvocation {
+                to: a.maddr,
+                method: MinerMethod::ControlAddresses as u64,
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: *REWARD_ACTOR_ADDR,
+                method: RewardMethod::ThisEpochReward as u64,
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: *STORAGE_POWER_ACTOR_ADDR,
+                method: PowerMethod::CurrentTotalPower as u64,
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: a.client1,
+                method: AccountMethod::AuthenticateMessage as u64,
+                params: Some(auth_params_ser),
```

Comment:
Could you add the expected ILLEGAL_ARGUMENT exit code here too?

Reply:
Certainly!